asserts/database.go

```diff
@@ -176,3 +223,110 @@ func (db *Database) Check(assert Assertion) error {
 	}
 	return fmt.Errorf("failed signature verification: %v", lastErr)
 }
+
+func (db *Database) checkAssertionRevisionAndClearLatest(indexPath string, newRev int) error {
```

Comment:
Sorry for not being more clear. There's clear functionality here which is "get me the current revision of this assertion". This should be in a function on its own.. the clear latest and replace is very specific behavior to the adding of the new assertion, and can remain in the original function.